tmd/view/view.py

```diff
@@ -1115,10 +1115,162 @@ def polar_plot(pop, neurite_type="neurites", bins=20):
     """Generate a polar plot of a neuron or population."""
     input_data = _get_polar_data(pop, neurite_type=neurite_type, bins=bins)
 
-    fig = _cm.plt.figure()
+    fig = plt.figure()
     ax = fig.add_axes([0.1, 0.1, 0.8, 0.8], polar=True)
 
-    theta = _np.array(input_data)[:, 0]
-    radii = _np.array(input_data)[:, 2] / _np.max(input_data)
-    width = 2 * _np.pi / len(input_data)
-    ax.bar(theta, radii, width=width, bottom=0.0, alpha=0.8)
+    theta = np.array(input_data)[:, 0]
+    norm_radii = np.array(input_data)[:, 2] / np.max(input_data)
+    width = 2 * np.pi / len(input_data)
+    ax.bar(theta, norm_radii, width=width, bottom=0.0, alpha=0.8)
+
+
+def tree_colors(
+    tr,
+    ph_graph,
+    colors=None,
+    new_fig=True,
+    subplot=111,
+    new_axes=True,
+    plane="xy",
+    cmap=plt.cm.jet,
+    **kwargs,
+):
+    """Generate a 2d pic of the tree, each branch has a unique color.
+
+    Generate a barcode using the corresponding colors.
+    """
+    if plane not in ("xy", "yx", "xz", "zx", "yz", "zy"):
+        return None, "No such plane found! Please select one of: xy, xz, yx, yz, zx, zy."
+
+    # Initialization of matplotlib figure and axes.
+    fig, ax = cm.get_figure(new_fig=new_fig, new_axes=new_axes, subplot=subplot)
+
+    # Initialization of colors to be used.
+    if colors is None:
+        ordered_nums = [np.float(i) / len(ph_graph) for i in np.arange(len(ph_graph))]
+        colors_random = cmap(ordered_nums)
+    else:
+        colors_random = colors
+
+    # Definition of tree branch colors based on persistence
+    def get_colors_ph(tr, ph_graph, colors):
+        """Assigns colors to each tree branch according to the persistence levels."""
+        beg, end = tr.get_sections_2()
+        end_graph = {}
+        for j, graph_id in enumerate(ph_graph):
+            for gi in graph_id:
+                end_id = np.where(end == gi)[0][0]
+                end_graph[end_id] = j
+
+        sec_ids = [1] * (end[0] - beg[0])
+        for i, e in enumerate(end[1:]):
+            sec_ids = sec_ids + [i + 2] * (e - end[i])
+
+        colors_select = [colors[end_graph[s - 1]] for s in sec_ids]
+        return colors_select
+
+    treecolors = get_colors_ph(tr, ph_graph, colors_random)
+
+    # Data needed for the viewer: x,y,z,r
+    bounding_box = tr.get_bounding_box()
+
+    def _seg_2d(seg):
+        """2d coordinates required for the plotting of a segment."""
+        horz = term_dict[plane[0]]
+        vert = term_dict[plane[1]]
+
+        horz1 = seg[0][horz]
+        horz2 = seg[1][horz]
+        vert1 = seg[0][vert]
+        vert2 = seg[1][vert]
+
+        return ((horz1, vert1), (horz2, vert2))
+
+    segs = [_seg_2d(seg) for seg in tr.get_segments()]
+
+    # Definition of the linewidth according to diameter, if diameter is True.
+
+    linewidth = [d * 1.0 for d in tr.d]
```

Comment:
Is this to convert the tr.d to a list? How about `list(tr.d)`

Reply:
modified, good point!